src/Roslyn.Diagnostics.Analyzers/Core/AbstractRunIterations`1.cs

```diff
@@ -128,5 +128,19 @@ private async Task<Document> RunIterationsAsync(Document document, TMethodDeclar
             var root = await method.SyntaxTree.GetRootAsync(cancellationToken);
             return document.WithSyntaxRoot(root.ReplaceNode(method, updatedMethod));
         }
+
+        private static SyntaxNode ReplaceName(SyntaxGenerator syntaxGenerator, SyntaxNode node, string name)
+        {
+            var newNode = syntaxGenerator.WithName(node, name);
+            if (newNode.RawKind != node.RawKind
+                && newNode.ChildNodes().FirstOrDefault()?.RawKind == node.RawKind)
```

Comment:
I’d personally prefer language specific implementations using explicit syntax kinds or syntax nodes, but it’s your call.